runtime/src/main/kotlin/kotlin/native/internal/KClassImpl.kt

```diff
@@ -87,3 +87,6 @@ private external fun getRelativeName(typeInfo: NativePtr): String?
 
 @SymbolName("Kotlin_TypeInfo_isInstance")
 private external fun isInstance(obj: Any, typeInfo: NativePtr): Boolean
+
+@TypedIntrinsic(IntrinsicType.INSTANCE_OF)
+internal external inline fun <reified T : Any> instanceOf(obj: Any): Boolean
```

Comment:
Placing this intrinsic to `Intrinsics.kt` would probably make more sense.
Also please add `@PublishedApi`, we follow convention of using it to indicate that `internal` declaration can be used from foreign module (which is the case here).